src/main/java/com/limechain/beefy/dto/Commitment.java

```diff
@@ -1,11 +1,15 @@
 package com.limechain.beefy.dto;
 
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import java.math.BigInteger;
 import java.util.List;
 
 @Data
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
```suggestion
@Value
```

and remove `private` modifiers.